src/binding/c/rma_api.txt

```diff
@@ -0,0 +1,466 @@
+    for this window. This set of hints may be greater or smaller than the set of
+    hints specified when the window was created, as the system may not recognize
+    some hints set by the user, and may recognize other hints that the user has not
+    set.
+*/
+
+MPI_Win_get_name:
+    .desc: Get the print name associated with the MPI RMA window
+{
+    MPL_strncpy(win_name, win_ptr->name, MPI_MAX_OBJECT_NAME);
+    *resultlen = (int) strlen(win_name);
+}
+
+MPI_Win_lock:
+    .desc: Begin an RMA access epoch at the target process
+    .skip: validate-ASSERT, validate-LOCK_TYPE
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+{ -- error_check --
+    CHECKMASK: assert, assert, MPI_MODE_NOCHECK
+    CHECKENUM: lock_type, locktype, MPI_LOCK_SHARED MPI_LOCK_EXCLUSIVE
+}
+
+MPI_Win_lock_all:
+    .desc: Begin an RMA access epoch at all processes on the given window
+    .seealso: MPI_Win_unlock_all
+    .skip: validate-ASSERT
+    .impl: mpid
+/*
+    Notes:
+    
+    This call is not collective.
+    
+    The 'assert' argument is used to indicate special conditions for the fence that
+    an implementation may use to optimize the 'MPI_Win_lock_all' operation.  The
+    value zero is always correct.  Other assertion values may be or''ed together.
+    Assertions that are valid for 'MPI_Win_lock_all' are\:
+    
+    . 'MPI_MODE_NOCHECK' - No other process holds, or will attempt to acquire a
+      conflicting lock, while the caller holds the window lock. This is useful
+      when mutual exclusion is achieved by other means, but the coherence
+      operations that may be attached to the lock and unlock calls are still
+      required.
+    
+    There may be additional overheads associated with using 'MPI_Win_lock' and
+    'MPI_Win_lock_all' concurrently on the same window. These overheads could be
+    avoided by specifying the assertion 'MPI_MODE_NOCHECK' when possible
+*/
+{ -- error_check --
+    CHECKMASK: assert, assert, MPI_MODE_NOCHECK
+}
+
+MPI_Win_post:
+    .desc: Start an RMA exposure epoch
+    .skip: ThreadSafe, validate-ASSERT
+    .impl: mpid
+{ -- error_check --
+    CHECKMASK: assert, assert, MPI_MODE_NOCHECK MPI_MODE_NOSTORE MPI_MODE_NOPUT
+}
+
+MPI_Win_set_info:
+    .desc: Set new values for the hints of the window associated with win
+    .seealso: MPI_Win_get_info
+    .impl: mpid
+/*
+    Notes:
+    
+    Some info items that an implementation can use when it creates a window cannot
+    easily be changed once the window has been created. Thus, an implementation may
+    ignore hints issued in this call that it would have accepted in a creation
+    call.
+*/
+
+MPI_Win_set_name:
+    .desc: Set the print name for an MPI RMA window
+{
+    MPL_strncpy(win_ptr->name, win_name, MPI_MAX_OBJECT_NAME);
+}
+
+MPI_Win_shared_query:
+    .desc: Query the size and base pointer for a patch of a shared memory window
+    .seealso: MPI_Win_allocate_shared
+    .impl: mpid
+
```

Comment:
In `win_shared_query`, The original description for output `baseptr` is useful. Can we keep it?
```
- baseptr - base pointer in the calling process'' address space of the shared
  segment belonging to the target rank.
```

Reply:
Parameter description is from  `src/binding/c/mpi_standard_api.txt`, which is kept in sync with MPI standard, and I think we should keep it this way.

We can add a note for any specifics that is missing from standard description or that we want to emphasize. Would you like me to do so?

Reply:
maybe we can move the description to the note paragraph? No need to hack the auto-sync file.

Reply:
> maybe we can move the description to the note paragraph? No need to hack the auto-sync file.

👍

Reply:
fixed